pilot/pkg/networking/core/v1alpha3/envoyfilter/listener_patch.go

```diff
@@ -430,31 +429,80 @@ func patchHTTPFilters(patchContext networking.EnvoyFilter_PatchContext,
 		}
 		patchHTTPFilter(patchContext, filterKey, patches, listener, fc, filter, httpFilter, &httpFiltersRemoved)
 	}
+	opatches := make([]*model.EnvoyFilterConfigPatchWrapper, 0)  // List of final patches to be processed.
+	mpatches := make([]*model.EnvoyFilterConfigPatchWrapper, 0)  // List of matched patches based on patch context.
+	cpatches := make([]*model.EnvoyFilterConfigPatchWrapper, 0)  // List of patches that have cyclic dependencies.
+	inpatches := make([]*model.EnvoyFilterConfigPatchWrapper, 0) // List of matched patches excluding cyclic dependencies.
+	dependents := map[*model.EnvoyFilterConfigPatchWrapper]struct{}{}
 	applied := false
-	for _, cp := range patches[networking.EnvoyFilter_HTTP_FILTER] {
-		if !commonConditionMatch(patchContext, cp) ||
-			!listenerMatch(listener, cp) ||
-			!filterChainMatch(listener, fc, cp) ||
-			!networkFilterMatch(filter, cp) {
+
+	// First collect all matched patches.
+	for _, lp := range patches[networking.EnvoyFilter_HTTP_FILTER] {
+		if !commonConditionMatch(patchContext, lp) ||
+			!listenerMatch(listener, lp) ||
+			!filterChainMatch(listener, fc, lp) ||
+			!networkFilterMatch(filter, lp) {
+			continue
+		}
+		mpatches = append(mpatches, lp)
+	}
+
```

Comment:
Note: n^2 loop, though expected patches are normally small.

Reply:
yes. That is what I was assuming when I coded this